src/CachePlugin.php

```diff
@@ -242,7 +268,7 @@ private function createCacheKey(RequestInterface $request)
      */
     private function getMaxAge(ResponseInterface $response)
     {
-        if (!$this->config['respect_cache_headers']) {
+        if (!$this->config['respect_cache_headers'] || !in_array('max-age', $this->config['respect_response_cache_directives'], true)) {
```

Comment:
i would do the cleanup (create an empty array in respect_response_cache_directives) of this in the constructor and issue a deprecation warning if respect_cache_headers is defined. if both are defined in the constructor, throw an exception as its wrong to do that.